### PR TITLE
Random trail generation

### PR DESCRIPTION
**Pull request by Rytisok**
Random trail generation added to API
Now hardcoded trail directions for each trail marker will be ignored, instead, a random trail, based on difficulty will be generated

How trail generation works:
-Trail center point is given to function and 2 random points around it are generated
-Google maps directions api connects first random point, center point, second random point and returns encoded direction polyline
-Direction polyline is decoded into geojson

### CODE REVIEW

File: Encounter Me.Api/Trail generation Utilities/GooglePoints.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            {
+                // calculate next latitude
+                sum = 0;
+                shifter = 0;
+                do
+                {
+                    next5bits = (int)polylineChars[index++] - 63;
+                    sum |= (next5bits & 31) << shifter;
+                    shifter += 5;
+                } while (next5bits >= 32 && index < polylineChars.Length);
```
Comment (GabijaG) [inline Encounter Me.Api/Trail generation Utilities/GooglePoints.cs:43]: could you explain what is going on here? Seems like you are trying to somehow modify the `polylineChars` char array, but I can't understand what you are trying to achieve.

File: Encounter Me.Api/Trail generation Utilities/GooglePoints.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+                if (index >= polylineChars.Length && next5bits >= 32)
+                    break;
```
Comment (GabijaG) [inline Encounter Me.Api/Trail generation Utilities/GooglePoints.cs:61]: Also, code seems to be duplicated - maybe the duplicating part could be a method of the `Coordinate` class?

File: Encounter Me.Api/Trail generation Utilities/GooglePoints.cs (hunk 1 above)
Comment (Rytisok) [reply Encounter Me.Api/Trail generation Utilities/GooglePoints.cs:43]: Probably should have mentioned this in the script, but this is code from: https://gist.github.com/shinyzhu/4617989
Google trails API returns encoded polyline, which needs to be decoded into actual geojson to be used, this script does just that.
Seems like in this exact part algorithm decodes the encoded polyline attributes with bit shifting.

File: Encounter Me.Api/Trail generation Utilities/GooglePoints.cs (hunk 2 above)
Comment (ZagreDo) [reply Encounter Me.Api/Trail generation Utilities/GooglePoints.cs:61]: I agree, we should imply on single responsibility rule more.

**Review by ZagreDo — APPROVED** [review]

**Pull request merged**